### ISSUE TITLE
Feature Request: Option to Select Screen for Fullscreen Mode with Multiple Monitors

### ISSUE THREAD

**Issue by DGINXREAL** [CorsixTH/CorsixTH#2731]
### What feature or enhancement would you like to add?

I would like to add an option in the settings menu to allow players to select which monitor the game will display on when in fullscreen mode. This feature should detect multiple monitors and provide a dropdown or similar interface to choose the desired display.

### How would this feature benefit the gameplay of CorsixTH?

This feature would greatly enhance the convenience and usability of the game for players with multi-monitor setups. It allows players to customize their gaming experience by selecting the most suitable screen for their playstyle or environment. Whether it's optimizing performance, arranging their workspace, or dedicating one screen to gameplay while keeping another for other tasks, this feature ensures CorsixTH remains adaptable to diverse user setups. This improvement would modernize the game’s compatibility with common multi-monitor configurations.

### Was the feature in the original game?

No

### What version of CorsixTH are you using?

0.68

### Does this feature/enhancement relate to a current problem?

No

**Comment by lewri** [comment]
You can currently achieve this by having CorsixTH open on the display you want to have it on, then toggle fullscreen.

I believe you're asking however, for an in-game setting that detects displays connected and set full screen on x display that way. That indeed would be a nice approach (and one i like seeing in some other games) but I know it's still not a de-facto standard. And at least we're better than some games that always target display 1 for fullscreen regardless of what you do.

**Comment by TheCycoONE** [comment]
I'm not going to implement this until after I port to SDL3. In the mean time mock ups/more description would be useful. I'm not familiar with this mechanic.

**Comment by lewri** [comment]
> I'm not going to implement this until after I port to SDL3. In the mean time mock ups/more description would be useful. I'm not familiar with this mechanic.

One example I have is the Paradox Launcher interface for Prison Architect. It asks what screen to use:
![image](https://github.com/user-attachments/assets/7ede04eb-6b43-4592-a404-5ef293f60d36)

In game, it also asks:
![image](https://github.com/user-attachments/assets/83acdc1a-336d-4c09-a90e-1c705e187fb0)

I'm sure we can do better labelling than the attempt above for displays (however, both my screens are identical so you can't tell them apart anyway).

**Comment by lewri** [comment]
As another example, Barotrauma also added this to their Winter 2024 update
![image](https://github.com/user-attachments/assets/5d39ad8d-3015-415b-875b-abdde7021ebd)

**Comment by DGINXREAL** [comment]
The examples thats @lewri provided is exactly that what i mean :)

**Comment by TheCycoONE** [comment]
Yep, this looks doable. Query the available indexes and names. Save the index to config. Might go funny if you plug in a different display configuration but that's expected.

**Comment by DGINXREAL** [comment]
Maybe you can hash the names or idk if u can fetch any UID of the displays, that will prevent issues if you change your displays. When change or displays missing reset to index 0

**Comment by TheCycoONE** [comment]
Could be done, I don't know if it's worth it vs just saving the index (there is no UID) and resetting to 0 if missing. You can check the options file for one of those games. That would be the only way to let you hand edit the config file, which is typically possible.

e.g. dosbox-staging does:
```
[sdl]
# display: Number of display to use; values depend on OS and user settings.
display            = 0
```

**Comment by lewri** [comment]
Config file in Barotrauma follows the same idea:

```yaml
  <graphicssettings
    display="0" #<-- this setting#
    width="1920"
    height="1080"
    vsync="false"
    compresstextures="true"
    framelimit="200"
    displaymode="BorderlessWindowed"
    particlelimit="1500"
    specularity="true"
    chromaticaberration="true"
    losmode="Transparent"
    hudscale="1"
    inventoryscale="1"
    lightmapscale="1"
    visiblelightlimit="50"
    textscale="1"
    radialdistortion="true" />
```

**Comment by ARGAMX** [comment]
Seems we moved to SDL3?
So, can we somehow determine the further fate of this task?
